src/main/kotlin/com/chattriggers/ctjs/api/render/Renderer.kt

```diff
@@ -738,12 +738,16 @@ object Renderer {
         matrixStack.pop()
     }
 
-    internal fun withMatrix(stack: MatrixStack, partialTicks: Float = Renderer.partialTicks, block: () -> Unit) {
+    internal fun withMatrix(stack: MatrixStack?, partialTicks: Float = Renderer.partialTicks, block: () -> Unit) {
         Renderer.partialTicks = partialTicks
         matrixPushCounter = 0
-        pushMatrix(UMatrixStack(stack))
+        if (stack != null)
+            pushMatrix(UMatrixStack(stack))
+
         block()
-        popMatrix()
+
+        if (stack != null)
+            popMatrix()
```

Comment:
Not super related to your changes, but I think `block` can execute user-code and thus throw exception, so the push-pop should probably be in a `try`/`finally` block. Might as well do it while you're here